docs/source/examples/beam.py

```diff
@@ -0,0 +1,46 @@
+from gpkit.shortcuts import *
+import numpy as np
+import matplotlib.pyplot as plt
+
+N = 100
+L = 5.
+dx = L/(N-1)
+ei = 1E4
+EI = Var("EI", ei)
+
+P = 100.
+p = Vec(N, "p", descr="Distributed load")
+p = p.sub(p, P*np.ones(N))
+
+V  = Vec(N, "V", label="Internal shear")
+M  = Vec(N, "M", label="Internal moment")
+th = Vec(N, "th", label="Slope")
+w  = Vec(N, "w", label="Displacement")
+
+eps = 1E-16 #something small
+
+substitutions = {var: eps for var in [V[-1], M[-1], th[0], w[0]]}
```

Comment:
It feels odd that this is needed but I see why -- keeping all the vector variables the same length. Let's let this slide for now but keep in mind that we may want to revisit this as we build up more examples that involve discretization.